cmd/kubeadm/app/phases/kubeconfig/kubeconfig.go

```diff
@@ -362,32 +362,39 @@ func ValidateKubeconfigsForExternalCA(outDir string, cfg *kubeadmapi.InitConfigu
 }
 
 func getKubeConfigSpecsBase(cfg *kubeadmapi.InitConfiguration) (map[string]*kubeConfigSpec, error) {
-	apiServer, err := kubeadmutil.GetControlPlaneEndpoint(cfg.ControlPlaneEndpoint, &cfg.LocalAPIEndpoint)
+	controlPlaneEndpoint, err := kubeadmutil.GetControlPlaneEndpoint(cfg.ControlPlaneEndpoint, &cfg.LocalAPIEndpoint)
```

Comment:
@fabriziopandini alternatively i can make GetControlPlaneEndpoint() return `controlPlaneEndpoint, localAPIEndpoint, error`

Reply:
I'm ok with the solution you are proposing.
Having `GetControlPlaneEndpoint` returning two endpoints seems a little bit weird to me...